components/api-resource-mgt/org.wso2.carbon.identity.api.resource.mgt/src/main/java/org/wso2/carbon/identity/api/resource/mgt/APIResourceManager.java

```diff
@@ -45,6 +45,23 @@ APIResourceSearchResult getAPIResources(String after, String before, Integer lim
                                             String tenantDomain)
             throws APIResourceMgtException;
 
+    /**
+     * Get API resources with required attributes.
+     *
+     * @param after              Get API resources after this value.
+     * @param before             Get API resources before this value.
+     * @param limit              Number of API resources to retrieve.
+     * @param filter             Filter expression.
+     * @param sortOrder          Sort order.
+     * @param tenantDomain       Tenant domain.
+     * @param requiredAttributes Required attributes.
+     * @return
+     * @throws APIResourceMgtException
```

Comment:
complete the comment

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/5020/commits/2121caf2f730580e7b8e199b87ecb6ba67bbed06